static/src/js/components/GranuleImage/GranuleImage.js

```diff
@@ -13,128 +13,99 @@ import './GranuleImage.scss'
-    super(props)
 
-    this.state = {
-      isOpen: true,
-      isLoaded: false,
-      isLoading: true,
-      isErrored: false
-    }
-
-    this.handleToggleImage = this.handleToggleImage.bind(this)
-    this.handleImageLoaded = this.handleImageLoaded.bind(this)
-    this.handleImageErrored = this.handleImageErrored.bind(this)
-  }
-
-  UNSAFE_componentWillReceiveProps(nextProps) {
-    const { imageSrc } = this.props
-    const { imageSrc: nextImageSrc } = nextProps
-    if (nextImageSrc && imageSrc !== nextImageSrc) {
-      this.setState({
-        isLoaded: false,
-        isLoading: true,
-        isErrored: false
-      })
-    }
-  }
-
-  handleToggleImage() {
-    const {
-      isOpen
-    } = this.state
-
-    this.setState({
-      isOpen: !isOpen
-    })
-  }
-
-  handleImageLoaded() {
-    this.setState({
-      isLoaded: true,
-      isLoading: false,
-      isErrored: false
-    })
-  }
-
-  handleImageErrored() {
-    this.setState({
-      isLoaded: false,
-      isLoading: false,
-      isErrored: true
-    })
+export const GranuleImage = ({
+  imageSrc
+}) => {
+  const [isOpen, setIsOpen] = useState(true)
+  const [isLoaded, setIsLoaded] = useState(false)
+  const [isLoading, setIsLoading] = useState(true)
+  const [isErrored, setIsErrored] = useState(false)
+
+  const containerClassName = classNames({
+    'granule-image': true,
+    'granule-image--is-open': isOpen
+  })
+
+  const imageClassName = classNames({
+    'granule-image__image': true,
+    'granule-image__image--is-loaded': isLoaded
+  })
+
+  const handleToggleImage = (() => {
+    setIsOpen(!isOpen)
+  })
+
+  const handleImageLoaded = (() => {
+    setIsLoaded(true)
+    setIsLoading(false)
+    setIsErrored(false)
+  })
+
+  const handleImageErrored = (() => {
+    setIsLoaded(false)
+    setIsLoading(false)
+    setIsErrored(true)
+  })
+
+  const loadImage = () => {
+    setIsErrored(false)
+    setIsLoaded(false)
+    setIsLoading(true)
   }
 
-  render() {
-    const {
-      isErrored,
-      isLoaded,
-      isLoading,
-      isOpen
-    } = this.state
-
-    const { imageSrc } = this.props
-
-    const containerClassName = classNames({
-      'granule-image': true,
-      'granule-image--is-open': isOpen
-    })
-
-    const imageClassName = classNames({
-      'granule-image__image': true,
-      'granule-image__image--is-loaded': isLoaded
-    })
-
-    if (!imageSrc) return null
-
-    return (
-      <div className={containerClassName}>
+  // When the imageSrc changes value then load the new image
+  useEffect(() => {
+    loadImage()
+  }, [imageSrc])
+
+  if (!imageSrc) return null
+
+  return (
+    <div className={containerClassName}>
+      { isOpen ? (
+        <Button
+          className="granule-image__button granule-image__button--close"
+          icon={FaTimes}
+          onClick={handleToggleImage}
+          label="Close browse image"
+          data-testid="granule-image__button granule-image__button--close"
```

Comment:
`granule-image__button` doesn't belong in the testid, think of it like a regular id attribute, its just used for testing

Reply:
removed